cmd/ops_agent_uap_plugin/service_windows.go

```diff
@@ -53,9 +189,257 @@ func (ps *OpsAgentPluginServer) Stop(ctx context.Context, msg *pb.StopRequest) (
 // plugins detect some non-fatal errors causing it unable to offer some features
 // it can reported in status which is sent back to the service by agent.
 func (ps *OpsAgentPluginServer) GetStatus(ctx context.Context, msg *pb.GetStatusRequest) (*pb.Status, error) {
-	panic("GetStatus method is not implemented on Windows yet")
+	log.Println("Received a GetStatus request")
+	ps.mu.Lock()
+	defer ps.mu.Unlock()
+	if ps.cancel == nil {
+		log.Println("The Ops Agent plugin is not running")
+		return &pb.Status{Code: 1, Results: []string{"The Ops Agent Plugin is not running."}}, nil
+	}
+	log.Println("The Ops Agent plugin is running")
+	return &pb.Status{Code: 0, Results: []string{"The Ops Agent Plugin is running ok."}}, nil
+}
+
+// serviceManager is an interface to abstract the Windows service manager. This is used to facilitate testing.
```

Comment:
Would it make sense to move this to a separate file in `utils`? I feel like it distracts from the core logic of this file.